verification/deriveChild.go

```diff
@@ -0,0 +1,288 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestDeriveChild(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var resp *DeriveChildResp
+
+	simulation := false
+	handle := getInitialContextHandle(d, c, t, simulation)
+	defer func() {
+		c.DestroyContext(handle, DestroyDescendants)
+	}()
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+
+	digestLen := profile.GetDigestSize()
+	currentLocality := d.GetLocality()
+	otherLocality := currentLocality + 1
+
+	// Child context handle returned will be the default context handle when MakeDefault flag is set
+	if resp, err = c.DeriveChild(handle, make([]byte, digestLen), MakeDefault, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while creating child context handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	if resp.NewContextHandle != DefaultContextHandle {
+		t.Fatalf("[FATAL]: Incorrect handle. Should return %v, but returned %v", DefaultContextHandle, *handle)
+	}
```

Comment:
This is made a separate test case and is moved out of list of common testcases - i.e. "AllTestCases" list

Reply:
Can you make a separate list (similar to AllTestCases) called something like `IrreversibleTestCases`. And add any test case to that list which permenantly modifies the DPE tree?

This will make it easier for a target (other than the simulator) with reset control to run these tests separately, resetting DPE in between each one.

Reply:
Added the list, so far there are just two tests DeriveChild and DeriveChild in Simulation mode that alter DPE tree.